filament/src/ShadowMap.cpp

```diff
@@ -428,32 +431,37 @@ void ShadowMap::computeShadowCameraDirectional(
             // We know we're using an ortho projection
             mTexelSizeWs = texelSizeWorldSpace(St.upperLeft());
         }
-        mLightSpace = St;
-
-        mLightSpaceVsm = computeVsmLightSpaceMatrix(mLightSpace, Mv, zfar);
+        if (!mShadowMapInfo.vsm) {
+            mLightSpace = St;
+        } else {
+            mLightSpace = computeVsmLightSpaceMatrix(St, Mv, zfar);
+        }
 
         // We apply the constant bias in world space (as opposed to light-space) to account
         // for perspective and lispsm shadow maps. This also allows us to do this at zero-cost
         // by baking it in the shadow-map itself.
-
-        const mat4f b = mat4f::translation(dir * params.options.constantBias);
-        const mat4f Sb = S * b;
+        const float constantBias = mShadowMapInfo.vsm ? 0.0f : params.options.constantBias;
+        const mat4f b = mat4f::translation(dir * constantBias);
 
         // It's important to set the light camera's model matrix separately from its projection, so
         // that the cameraPosition uniform gets set correctly.
         // mLightSpace is used in the shader to access the shadow map texture, and has the model
         // matrix baked in.
 
-        mCamera->setModelMatrix(FCamera::rigidTransformInverse(b) * M);
+        // The model matrix below is in fact inverted to get the view matrix and passed to the
+        // shader as 'viewFromWorldMatrix', and is used in the VSM case to compute the depth metric.
+        // (see depth_main.fs). Note that in the case of VSM, 'b' below is identity.
+        mCamera->setModelMatrix(inverse(Mv * b));
```

Comment:
Can we not use `rigidTransformInverse` here?

Reply:
Yes, good point.